### ISSUE TITLE
Buildpack exception not canceling build

### ISSUE THREAD

**Issue by jorihardman** [deis/deis#3211]
I have an exception occurring when pushing an app using heroku-buildpack-ruby. The buildpack process throws and error, and yet the build goes through and deploys the app.

This is in a newly provisioned 1.4.0 cluster.

**Comment by bacongobbler** [comment]
@jorihardman do you have logs? Sounds related to https://github.com/deis/deis/pull/3207.

**Comment by jorihardman** [comment]
Unfortunately I just cleared my console and lost the logs. The last buildpack messages were a ruby stack trace. My particular error was caused by my database url being wrapped in single quotes, causing the db connection to fail with BadURI error.

**Comment by bacongobbler** [comment]
I think I saw it on IRC:

```
URI::InvalidURIError: bad URI(is not URI?): 'mysql2://XXXXXX:XXXXXX@1.2.3.4:3306/swipesense'
```

So yes that should be solved by #3207 (and the underlying issue in #3213)
